src/test/java/uk/gov/hmcts/reform/sscs/services/ccd/CcdCasesSenderTest.java

```diff
@@ -253,6 +270,77 @@ public void shouldAddExistingNewHearingDetailsFromCcdToCaseWhenNoHearingDetailsi
+
+        assertThat(caseDataArgumentCaptor.getValue().getRegion(), equalTo(regionalProcessingCenter.getName()));
+        assertThat(caseDataArgumentCaptor.getValue().getRegionalProcessingCenter(), equalTo(regionalProcessingCenter));
+
+    }
+
+
+    @Test
+    public void shouldAddRegionalProcessingCenterForAnExistingCaseIfItsNotAlreadyPresentInCcd() throws Exception {
+        RegionalProcessingCenter regionalProcessingCenter = getRegionalProcessingCenter();
+        ArgumentCaptor<CaseData> caseDataArgumentCaptor = ArgumentCaptor.forClass(CaseData.class);
+
+        CaseData caseData = buildCaseData(RESPONSE_RECEIVED);
+        when(regionalProcessingCenterService.getByScReferenceCode("SC068/17/00011"))
+            .thenReturn(regionalProcessingCenter);
+
+        CaseDetails existingCaseDetails = getCaseDetails(CASE_DETAILS_WITH_HEARINGS_JSON);
+
+        ccdCasesSender.sendUpdateCcdCases(caseData, existingCaseDetails, idamTokens);
+
+        verify(updateCcdService).update(caseDataArgumentCaptor.capture(),
+            eq(existingCaseDetails.getId()), eq(caseData.getLatestEventType()), eq(idamTokens));
+
+        assertThat(caseDataArgumentCaptor.getValue().getRegionalProcessingCenter(), equalTo(regionalProcessingCenter));
+        assertThat(caseDataArgumentCaptor.getValue().getRegion(), equalTo(regionalProcessingCenter.getName()));
+    }
+
+    @Test
+    public void shouldNotAddRegionalProcessingCenterForAnExistingCaseIfItsAlreadyPresent() throws Exception {
+        RegionalProcessingCenter regionalProcessingCenter = getRegionalProcessingCenter();
+        ArgumentCaptor<CaseData> caseDataArgumentCaptor = ArgumentCaptor.forClass(CaseData.class);
+
+        CaseData caseData = buildCaseDataForEventAndCaseReference(RESPONSE_RECEIVED, "SC068/17/00013");
+        when(regionalProcessingCenterService.getByScReferenceCode("SC068/17/00013"))
+            .thenReturn(regionalProcessingCenter);
+
+        CaseDetails existingCaseDetails = getCaseDetails(CASE_DETAILS_WITH_NO_HEARINGS_JSON);
+
+        ccdCasesSender.sendUpdateCcdCases(caseData, existingCaseDetails, idamTokens);
+
+        verify(updateCcdService).update(caseDataArgumentCaptor.capture(),
+            eq(existingCaseDetails.getId()), eq(caseData.getLatestEventType()), eq(idamTokens));
+        verify(regionalProcessingCenterService, never()).getByScReferenceCode("SC068/17/00013");
+
+    }
+
+
+    private RegionalProcessingCenter getRegionalProcessingCenter() {
+        RegionalProcessingCenter regionalProcessingCenter = new RegionalProcessingCenter();
+        regionalProcessingCenter.setName("CARDIFF");
+        regionalProcessingCenter.setAddress1("HM Courts & Tribunals Service");
+        regionalProcessingCenter.setAddress2("Social Security & Child Support Appeals");
+        regionalProcessingCenter.setAddress3("Eastgate House");
+        regionalProcessingCenter.setAddress3("Newport Road");
```

Comment:
Is this supposed to be Address3 or Address4? if it is Address4, please, don't change it I will do it since I am doing some minor refactoring.

Reply:
Its address4.